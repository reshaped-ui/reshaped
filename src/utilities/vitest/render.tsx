import { render as vitestRender } from "vitest-browser-react";
import Reshaped from "components/Reshaped";
import "themes/reshaped/theme.css";

const Wrapper = (props: { children: React.ReactNode }) => {
	return <Reshaped>{props.children}</Reshaped>;
};

const render = (node: React.ReactNode) => {
	const screen = vitestRender(<div id="root">{node}</div>, {
		wrapper: Wrapper,
	});

	return { ...screen, container: document.getElementById("root")! };
};

export default render;
