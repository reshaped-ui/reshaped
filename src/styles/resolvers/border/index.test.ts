import { expect, test, describe } from "vitest";
import getBorderStyles from "./index";

describe("Styles/Border", () => {
	test("handles value", () => {
		expect(getBorderStyles("primary")).toMatchSnapshot();
	});

	test("handles undefined value", () => {
		expect(getBorderStyles()).toMatchSnapshot();
	});

	test("handles responsive value", async () => {
		expect(getBorderStyles({ s: undefined, m: "neutral", l: "transparent" })).toMatchSnapshot();
	});
});
