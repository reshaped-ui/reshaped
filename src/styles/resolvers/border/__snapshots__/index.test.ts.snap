// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Styles/Border > handles responsive value 1`] = `
{
  "classNames": [
    "_root_5fd06f",
    "_null_5fd06f",
    "_--border-neutral--m_5fd06f",
    "_--border-transparent--l_5fd06f",
  ],
}
`;

exports[`Styles/Border > handles undefined value 1`] = `null`;

exports[`Styles/Border > handles value 1`] = `
{
  "classNames": [
    "_root_5fd06f",
    "_--border-primary_5fd06f",
  ],
}
`;
